### PR TITLE
Refactor/streaming v4

### PR DESCRIPTION
**Pull request by LaurenSpiegel**

### CODE REVIEW

**Comment by LaurenSpiegel** [comment]
@ironman-machine go

File: tests/functional/jaws/src/test/java/com/scality/StreamV4AuthTest.java [hunk 1]
```diff
@@ -131,15 +132,24 @@ public void testStreamV4Auth() throws Exception {
...
     private static File createSampleFile(Integer fileSize) throws Exception {
+        String alph = "\r\nABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
+        SecureRandom random = new SecureRandom();
+        Integer randomLimit = Math.min(Math.round(fileSize/10), 10000);
+        StringBuilder myStringBuilder = new StringBuilder(randomLimit);
+        for(int i = 0; i < randomLimit; i++)
+              myStringBuilder.append(alph.charAt(random.nextInt(alph.length())));
+        String myString = myStringBuilder.toString();
         RandomAccessFile file = new RandomAccessFile(fileName, "rw");
+        file.writeUTF("\r\nlet's add some \r\n data with \r\n\rn\rn\rn\r\n\r\n\n");
```
Comment (bennettbuchanan) [inline tests/functional/jaws/src/test/java/com/scality/StreamV4AuthTest.java]: Some of these newlines (e.g., `\rn`) are not escaped. Is that intentional?

Comment (LaurenSpiegel) [inline tests/functional/jaws/src/test/java/com/scality/StreamV4AuthTest.java]: I want line breaks

**Comment by LaurenSpiegel** [comment]
end to end passed here: http://ci.ironmann.io/gh/scality/Integration/8443

**Pull request merged**